models/foreignreference.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2022 Gitea. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"code.gitea.io/gitea/models/db"
+)
+
+// ForeignTypeIssue is to be used for the Type field in ForeignReference
+const ForeignTypeIssue = "issue"
+
+// ForeignReference represents external references
+type ForeignReference struct {
+	// RepoID is the first column in all indices. now we only need 2 indices: (repo, local) and (repo, foreign, type)
+	RepoID    int64  `xorm:"UNIQUE(repo_foreign_type) INDEX(repo_local)" `
+	LocalID   int64  `xorm:"INDEX(repo_local)"` // the resource key inside Gitea, it can be IssueIndex, or some model ID.
+	ForeignID string `xorm:"INDEX UNIQUE(repo_foreign_type)"`
+	Type      string `xorm:"VARCHAR(8) INDEX UNIQUE(repo_foreign_type)"`
```

Comment:
Is 8 enough? In fact, we can enumerate them now.
`issue`, `pull_request`, `comment`, `review`, `review_comemnt`, `release`.

Maybe 16 is better?

Reply:
I added all types as suggested and increased the Type field size from 8 to 16.

---
[source](https://lab.fedeproxy.eu/forgefriends/forgefriends/-/merge_requests/36#note_1077)